p2p/module.go

```diff
@@ -0,0 +1,193 @@
+package p2p
+
+import (
+	"errors"
+	"log"
+	"net"
+	"pocket/p2p/types"
+	"pocket/shared/config"
+	"pocket/shared/modules"
+	"pocket/shared/types"
+	"sync"
+
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	MaxInbound           uint = 128
+	MaxOutbound          uint = 128
+	WireByteHeaderLength int  = 9
+	ReadBufferSize       int  = (1024 * 4)
+	WriteBufferSize      int  = (1024 * 4)
+	ReadDeadlineMs       int  = 400
+)
+
+var (
+	Protocol = "tcp"
+	Address  = "localhost:30303"
+)
+
+type P2PModule struct {
+	modules.NetworkModule
+
+	bus    modules.Bus
+	config *config.P2PConfig
+	c      *dcodec
+
+	id           uint64
+	protocol     string
+	address      string
+	externaladdr string
+
+	inbound  pipemap
+	outbound pipemap
+
+	peerlist *types.Peerlist
+
+	listener struct {
+		sync.Mutex
+		*net.TCPListener
+	}
+	listening atomic.Bool
+
+	done    chan uint
+	ready   chan uint
+	closed  chan uint
+	errored chan uint
+
+	sink     chan types.Work
+	handlers map[types.PeerEvent][]func(...interface{})
+
+	logger struct {
+		sync.RWMutex
+		print func(...interface{}) (int, error)
+	}
+
+	err struct {
```

Comment:
Question: are these tiny embedded structs a common pattern? I would have personally externalized them but also don't have as much experience with Go.

Reply:
I suppose they are, seen them quite a bit for small structs that won't be reused much outside of a file. I always use them when declaring a type is too cumbersome. I also mostly declare structs as types when I intend to add behavior to them and when their size and responsibility is relatively significant.